website/static/js/share/sideBar.js

```diff
@@ -0,0 +1,92 @@
+var $ = require('jquery');
+var m = require('mithril');
+var $osf = require('js/osfHelpers');
+var utils = require('./utils');
+
+var SideBar = {};
+
+SideBar.view = function(ctrl){
+    if (ctrl.vm.results === null){
+        return [];
+    }
+    return m('', [
+            m('.sidebarHeader',  ['Active filters:',
+              (ctrl.vm.optionalFilters.length > 0 || ctrl.vm.requiredFilters.length > 0) ? m('a', {
+                  style: {
+                      'float': 'right'
+                  }, onclick: function(event){
+                      ctrl.vm.optionalFilters = [];
+                      ctrl.vm.requiredFilters = [];
+                      utils.search(ctrl.vm);
+                    }
+              }, ['Clear ', m('i.fa.fa-close')]) : []]),
+            m('ul', {style:{'list-style-type': 'none', 'padding-left': 0}}, ctrl.renderFilters()),
+            m('.sidebarHeader', 'Providers:'),
+            m('ul', {style:{'list-style-type': 'none', 'padding-left': 0}}, ctrl.renderProviders()),
+    ]);
+};
+
+SideBar.controller = function(vm) {
+    var self = this;
+    self.vm = vm;
+
+    self.vm.sort = m.prop($osf.urlParams().sort || 'Relevance');
+    self.vm.requiredFilters = $osf.urlParams().required ? $osf.urlParams().required.split('|') : [];
+    self.vm.optionalFilters = $osf.urlParams().optional ? $osf.urlParams().optional.split('|') : [];
+
+    self.vm.sortMap = {
+        'Date': 'dateUpdated',
+        Relevance: null
+    };
+
+    self.renderSort = function(){
+        return $.map(Object.keys(self.vm.sortMap), function(a) {
+            return m('li',
+                m('a', {
+                    'href': '#',
+                    onclick: function(event) {
+                        self.vm.sort(a);
+                        utils.search(self.vm);
+                    }
+                }, a));
+        });
+    };
+
+    self.renderFilters = function(){
+        return $.map(self.vm.optionalFilters.concat(self.vm.requiredFilters), function(filter){
+            return m('li.renderFilter', [
+                m('a', {
+                    onclick: function(event){
+                        utils.removeFilter(self.vm, filter);
+                    }
+                }, [m('i.fa.fa-close'), ' ' + filter
+                ])
+            ]);
+        });
+    };
+
+    self.renderProviders = function () {
+        return $.map($.map(Object.keys(self.vm.ProviderMap), function(result, index){
```

Comment:
For readability, can you break this up into multiple variables? You might also break this up into multiple methods, which would make testing and refactoring easier.